src/lib/gadgets/foreign-field.unit-test.ts

```diff
@@ -56,11 +74,57 @@ for (let F of fields) {
 
   eq2(F.add, (x, y) => ForeignField.add(x, y, F.modulus), 'add');
   eq2(F.sub, (x, y) => ForeignField.sub(x, y, F.modulus), 'sub');
+  eq2(F.mul, (x, y) => ForeignField.mul(x, y, F.modulus), 'mul');
+  equivalentProvable({ from: [f], to: f })(
+    (x) => F.inverse(x) ?? throwError('no inverse'),
+    (x) => ForeignField.inv(x, F.modulus),
+    'inv'
+  );
+  eq2(
+    (x, y) => F.div(x, y) ?? throwError('no inverse'),
+    (x, y) => ForeignField.div(x, y, F.modulus),
+    'div'
+  );
+
+  // tests with inputs that aren't reduced mod f
+  let big264 = unreducedForeignField(264, F); // this is the max size supported by our range checks / ffadd
+  let big258 = unreducedForeignField(258, F); // rough max size supported by ffmul
+
+  equivalentProvable({ from: [big264, big264], to: big264 })(
+    F.add,
+    (x, y) => ForeignField.add(x, y, F.modulus),
+    'add unreduced'
+  );
+  // subtraction doesn't work with unreduced y because the range check on the result prevents x-y < -f
+  equivalentProvable({ from: [big264, f], to: big264 })(
+    F.sub,
+    (x, y) => ForeignField.sub(x, y, F.modulus),
+    'sub unreduced'
+  );
+  equivalentProvable({ from: [big258, big258], to: f })(
+    F.mul,
+    (x, y) => ForeignField.mul(x, y, F.modulus),
+    'mul unreduced'
+  );
+  equivalentProvable({ from: [big258], to: f })(
+    (x) => F.inverse(x) ?? throwError('no inverse'),
+    (x) => ForeignField.inv(x, F.modulus),
+    'inv unreduced'
+  );
+  // the div() gadget doesn't work with unreduced x because the backwards check (x/y)*y === x fails
```

Comment:
It may be useful to document this at the interface level, i.e., what is compatible with unreduced values